os/kernel/pthread/pthread_mutexunlock.c

```diff
@@ -117,16 +146,24 @@
 
 int pthread_mutex_unlock(FAR pthread_mutex_t *mutex)
 {
-	int ret = EINVAL;
+	int ret = OK;
 
 	svdbg("mutex=0x%p\n", mutex);
 	DEBUGASSERT(mutex != NULL);
+	if (mutex == NULL) {
+		return EINVAL;
+	}
 
 	/* Make sure the semaphore is stable while we make the following checks.
 	 * This all needs to be one atomic action.
 	 */
 	sched_lock();
-	if (mutex != NULL) {
+
+	/* The unlock operation is only performed if the mutex is actually locked.
+	 * If the mutex is not locked, then SUCCESS will be returned (there is
+	 * no error return value specified for this case).
+	 */
+	if (pthread_mutex_islocked(mutex)) {
```

Comment:
Hi, As per the pthread.h specification,  2 conditions should be met. Refer line 106, 111 & 118 .
PTHREAD_MUTEX_NORMAL :  If any thread attempts to unlock an unlocked mutex results in undefined behavior and error no is not specified
PTHREAD_MUTEX_ERRORCHECK :  It should return with an ERROR 
PTHREAD_MUTEX_RECURSIVE :  It should return with an ERROR 

returning SUCCESS may break the pthread specification.